ItemPool.py

```diff
@@ -850,10 +850,16 @@ def get_pool_core(world: World) -> tuple[list[str], dict[str, Item]]:
                     dungeon_collection = dungeon.boss_key
                     if shuffle_setting == 'vanilla':
                         shuffle_item = False
-            # Map or Compass
-            elif location.vanilla_item in (dungeon.item_name("Map"), dungeon.item_name("Compass")):
-                shuffle_setting = world.settings.shuffle_mapcompass
-                dungeon_collection = dungeon.dungeon_items
+            # Map
+            elif location.vanilla_item in dungeon.item_name("Map"):
```

Comment:
This technically checks whether the item name is a substring of the dungeon's map item name. We don't have any item names that are substrings of each other like that, so this doesn't actually break anything, but a direct comparison is still more correct. Same with compasses below.

```suggestion
            elif location.vanilla_item == dungeon.item_name("Map"):
```